<?xml version="1.0"?>
<launch>
  <arg name="width"     default="640"/>
  <arg name="height"    default="360"/>
  <arg name="frame_rate" default="30"/>

  <group ns="d435">
    <node pkg="nodelet" type="nodelet" name="realsense2_camera_manager" args="manager" output="screen"/>
    <node pkg="nodelet" type="nodelet" name="realsense2_camera" args="load realsense2_camera/RealSenseNodeFactory realsense2_camera_manager">
      <param name="enable_pointcloud"        type="bool" value="false"/>
      <param name="enable_sync"              type="bool" value="false"/>
      <param name="align_depth"              type="bool" value="false"/>
      <param name="enable_fisheye"           type="bool" value="false"/>
      <param name="enable_imu"               type="bool" value="false"/>
  
      <param name="enable_depth"             type="bool" value="true"/>
      <param name="enable_infra1"            type="bool" value="false"/>
      <param name="enable_infra2"            type="bool" value="false"/>
      <param name="enable_color"             type="bool" value="false"/>
  
      <param name="depth_width"              type="int"  value="$(arg width)"/>
      <param name="depth_height"             type="int"  value="$(arg height)"/>
      <param name="depth_fps"                type="int"  value="$(arg frame_rate)"/>
      <param name="base_frame_id"            type="str"  value="d435_link"/>
      <param name="depth_frame_id"           type="str"  value="d435_depth_link"/>
      <param name="depth_optical_frame_id"   type="str"  value="d435_depth_optical"/>
    </node>
  </group>

  <node pkg="opengl_ros" type="depth_image_projector"
    name="depth_image_projector" output="screen">
    <param name="depthWidth"      value="$(arg width)"/>
    <param name="depthHeight"     value="$(arg height)"/>
    <param name="vertex_shader"   value="$(find opengl_ros_lib)/shader/vs_depth_image_projection.glsl"/>
    <param name="fragment_shader" value="$(find opengl_ros_lib)/shader/fs_depth_image_projection.glsl"/>

    <remap from="image_in"  to="/d435/depth/image_rect_raw"/>
    <remap from="image_out" to="/image_out"/>
  </node>

  <node pkg="image_view" type="image_view" name="viewer_in">
    <remap from="image" to="/d435/depth/image_rect_raw"/>
  </node>

  <node pkg="image_view" type="image_view" name="viewer_out">
    <remap from="image" to="/image_out"/>
  </node>
</launch>
